modules/dynamic_backends/main.tf

```diff
@@ -99,7 +100,7 @@ resource "google_compute_backend_service" "default" {
   connection_draining_timeout_sec = lookup(each.value, "connection_draining_timeout_sec", null)
   enable_cdn                      = lookup(each.value, "enable_cdn", false)
   security_policy                 = var.security_policy
-  health_checks                   = [google_compute_health_check.default[each.key].self_link]
+  health_checks                   = lookup(each.value, "health_check", null) ? null : [google_compute_health_check.default[each.key].self_link]
```

Comment:
I think this may fail when  `health_check` actually exists. `lookup` will return the whole `health_check` obj instead of a bool. We can do something like 
```hcl
lookup(each.value, "health_check", null) == null ? null : [google_compute_health_check.default[each.key].self_link]
```
to coerce this to bool.

Reply:
Thanks @bharathkkb, I've applied your suggestions.